desktop/sources/scripts/clock.js

```diff
@@ -60,19 +60,27 @@ function Clock (client) {
     console.log('Clock', 'Play', msg)
     if (this.isPaused === false) { return }
     this.isPaused = false
-    if (this.isPuppet === true) { console.warn('Clock', 'External Midi control'); return }
-    if (msg === true) { client.io.midi.sendClockStart() }
-    this.setSpeed(this.speed.target, this.speed.target, true)
+    if (this.isPuppet === true) { 
+      console.warn('Clock', 'External Midi control')
+      pulse.count = 0  // works in conjunction with `tap` advancing on 0
+      this.setFrame(0)  // make sure frame aligns with pulse count for an accurate beat
+    } else {
+      if (msg === true) { client.io.midi.sendClockStart() }
+      this.setSpeed(this.speed.target, this.speed.target, true)
+    }
   }
 
   this.stop = function (msg = false) {
     console.log('Clock', 'Stop')
     if (this.isPaused === true) { return }
     this.isPaused = true
-    if (this.isPuppet === true) { console.warn('Clock', 'External Midi control'); return }
-    if (msg === true || client.io.midi.isClock) { client.io.midi.sendClockStop() }
+    if (this.isPuppet === true) { 
+      console.warn('Clock', 'External Midi control')
+    } else {
+      if (msg === true || client.io.midi.isClock) { client.io.midi.sendClockStop() }
+      this.clearTimer()
+    }
     client.io.midi.allNotesOff()
-    this.clearTimer()
     client.io.midi.silence()
```

Comment:
I first thought these two lines should go inside the `else` branch, but I see they also need to be executed when Orca runs in Puppeteering mode to make sure all midi output is stopped 👍

Reply:
Ah, this was the additional improvement (stopping notes while in clock slave mode). Nice!

Reply:
I was trying to preserve existing `clearTimer` logic without fully knowing what it does, while making `allNotesOff` and `silence` fire in all cases, since we definitely need that during puppeteering.